metricflow/specs/where_filter_time_dimension.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import List, Sequence
+
+from dbt_semantic_interfaces.call_parameter_sets import FilterCallParameterSets, TimeDimensionCallParameterSet
+from dbt_semantic_interfaces.naming.dundered import DunderedNameFormatter
+from dbt_semantic_interfaces.protocols.protocol_hint import ProtocolHint
+from dbt_semantic_interfaces.references import EntityReference, TimeDimensionReference
+from dbt_semantic_interfaces.type_enums.time_granularity import TimeGranularity
+from typing_extensions import override
+
+from metricflow.specs.column_assoc import ColumnAssociationResolver
+from metricflow.specs.query_interface import QueryInterfaceDimension, QueryInterfaceTimeDimensionFactory
+from metricflow.specs.specs import TimeDimensionSpec
+
+
+class WhereFilterTimeDimension(ProtocolHint[QueryInterfaceDimension]):
+    """A time dimension that is passed in through the where filter parameter."""
+
+    @override
+    def _implements_protocol(self) -> QueryInterfaceDimension:
+        return self
+
+    def __init__(self, column_name: str):  # noqa
+        self.column_name = column_name
+
+    def grain(self, _grain: str) -> WhereFilterTimeDimension:
+        """The time granularity."""
+        raise NotImplementedError
+
+    def alias(self, _alias: str) -> WhereFilterTimeDimension:
+        """Renaming the column."""
+        raise NotImplementedError
+
+    def __str__(self) -> str:
+        """Returns the column name.
+
+        Important in the Jinja sandbox.
+        """
+        return self.column_name
+
+
+class WhereFilterTimeDimensionFactory(ProtocolHint[QueryInterfaceTimeDimensionFactory]):
+    """Creates a WhereFilterTimeDimension."""
+
+    @override
+    def _implements_protocol(self) -> QueryInterfaceTimeDimensionFactory:
+        return self
+
+    def __init__(  # noqa
+        self,
+        call_parameter_sets: FilterCallParameterSets,
+        column_association_resolver: ColumnAssociationResolver,
+    ):
+        self._call_parameter_sets = call_parameter_sets
+        self._column_association_resolver = column_association_resolver
+        self.time_dimension_specs: List[TimeDimensionSpec] = []
```

Comment:
Might be worth an update to the docstrings on these classes that the specs themselves are updated by reference with each call to create() and therefore are part of the public interface on the class.

It's in the code here in this PR, but it's not clear from looking at this module alone that this is the expected behavior and usage.